vllm_spyre/v1/worker/spyre_input_batch.py

```diff
@@ -174,23 +177,39 @@ def req_ids(self) -> list[str]:
         # while performing state updates to the batch.
         return cast(list[str], self._req_ids)
 
+    def get_available_index(self):
+        available_indices = self.req_indices_mask.logical_not().nonzero()
+        available_indices_list = available_indices.squeeze(dim=-1).tolist()
+        return available_indices_list[0] if available_indices_list else None
+
+    def req_idx_to_dense_index(self, req_index):
+        return self.req_indices_mask[:req_index].sum().item()
+
+    def req_id_to_dense_index(self, req_id):
+        req_index = self.req_id_to_index[req_id]
+        return self.req_idx_to_dense_index(req_index)
```

Comment:
It would really help if we can add some good detailed function docs to these functions, ideally with a very basic example to demonstrate what exactly it's returning (if applicable)

Reply:
Also adding the return types from these functions would be great too